ShoppingM8-be/src/main/java/edu/pw/shoppingm8/product/ProductService.java

```diff
@@ -0,0 +1,20 @@
+package edu.pw.shoppingm8.product;
+
+import java.util.List;
+
+import edu.pw.shoppingm8.product.api.dto.ProductCreateRequestDto;
+import edu.pw.shoppingm8.product.api.dto.ProductPatchRequestDto;
+
+public interface ProductService {
+    List<Product> getProducts(Long listId);
+
+    Product getProduct(Long listId, Long productId);
+
+    void purchaseProduct(Long listId, Long productId, Double amountChange);
+
+    Product createProduct(Long listId, ProductCreateRequestDto productDto);
+
+    void patchProduct(Long listId, Long productId, ProductPatchRequestDto productDto);
```

Comment:
"patch" is a HTTP method. In terms of business logic it's an edition method. It would be nice if it's name carried that information.

Reply:
fixed